test/unit/eriksen.spec.js

```diff
@@ -0,0 +1,141 @@
+'use strict';
+
+const chai = require('chai');
+const expect = chai.expect;
+const sinon = require('sinon');
+const proxyquire = require('proxyquire').noCallThru();
+
+chai.use(require('sinon-chai'));
+
+describe('Eriksen marshaler', function() {
+
+  let ProxyMock
+    , Eriksen
+    , marshal;
+
+  beforeEach(function() {
+    ProxyMock = sinon.stub();
+    Eriksen = proxyquire('../../../eriksen', { './lib/proxy': ProxyMock });
+    marshal = new Eriksen('burgers');
+  });
+
+  it('should set up with default values and config', function() {
+    expect(marshal.name).to.equal('burgers');
+    expect(marshal.models).to.be.an.instanceof(Map);
+    expect(marshal.config).to.be.an('object');
+    expect(marshal.config.primary).to.be.null;
+    expect(marshal.config.logger.log).to.be.a('function');
+    expect(marshal.config.logger.info).to.be.a('function');
+    expect(marshal.config.logger.error).to.be.a('function');
+  });
+
+  it('should add models by name', function() {
+    const modelA = {};
+    const modelB = {};
+    marshal.addModel('a', modelA);
+    marshal.addModel('b', modelB);
+
+    expect(marshal.models.size).to.equal(2);
+    expect(marshal.models.get('a')).to.equal(modelA);
+    expect(marshal.models.get('b')).to.equal(modelB);
+  });
+
+  it('should only add a model once, preferring the latest addition', function() {
+    const modelA1 = {};
+    const modelA2 = {};
+
+    marshal.addModel('a', modelA1);
+    marshal.addModel('a', modelA2);
+
+    expect(marshal.models.size).to.equal(1);
+    expect(marshal.models.get('a')).to.equal(modelA2);
+  });
+
+  it('should configure and set up a proxy', function() {
+    marshal.addModel('a', {});
+    marshal.configure({
+      primary: 'a',
+      secondary: false
+    });
+
+    expect(ProxyMock).to.have.been.calledWith({
+      models: marshal.models,
+      primary: 'a',
+      secondary: false,
+      logger: marshal.config.logger
+    });
+  });
+
+  it.skip('should be able to switch primary and secondary models', function() {
```

Comment:
I'm going to remove these two skipped tests, but just to point out that I had implemented a `marshal.switch()` method that handled switching primary and secondary models around for you, but I realized there'd be no reason to ever do that because the proxy you exported wouldn't get changed so we'll have to do this with config and restart, I am pretty sure. Just for the record.